src/lib.rs

```diff
@@ -308,8 +312,9 @@ impl Context {
             }
         }
 
-        let mut gid = ffi::ibv_gid::default();
-        let ok = unsafe { ffi::ibv_query_gid(ctx, PORT_NUM, 0, &mut gid as *mut _) };
+        // let mut gid = ffi::ibv_gid::default();
+        let mut gid = Gid::default();
+        let ok = unsafe { ffi::ibv_query_gid(ctx, PORT_NUM, 0, gid.as_mut() as _) };
```

Comment:
I _think_ this can now just be:
```suggestion
        let ok = unsafe { ffi::ibv_query_gid(ctx, PORT_NUM, 0, &mut gid) };
```
At least that was my hope with implementing `AsRef` and `AsMut`.

Reply:
As it turns out, you can convert `gid.as_mut() as _` to `gid.as_mut()`, but rust is not yet smart enough to figure out the types for `&mut gid`.